packages/editor/src/components/block-settings-menu/reusable-block-delete-button.js

```diff
@@ -35,9 +35,16 @@ export default compose( [
 			getBlock,
 			__experimentalGetReusableBlock: getReusableBlock,
 		} = select( 'core/editor' );
+		const { canUser } = select( 'core' );
+
 		const block = getBlock( clientId );
+
+		const id = block && isReusableBlock( block ) ? block.attributes.ref : null;
 		return {
-			reusableBlock: block && isReusableBlock( block ) ? getReusableBlock( block.attributes.ref ) : null,
+			id,
+			isDisabled: !! id && (
```

Comment:
I think same comment here regarding disabling vs. hiding.

Reply:
Also noting that in testing I didn't see the flash of this switching from enabled to disabled. That seems to be because the GET request for the block has already been made and cached. If an OPTIONS request was made instead of a GET request the issue would resurface.

Reply:
Have changed this to remove the button instead of disabling in 708a50484fc99537a5f3368b4dbb66e134198741.

We can't preload these requests because we don't know the reusable block ID ahead of time.

Reply:
Yep, I do understand that we can't preload. My worry is that it's only by coincidence that the user's capability is known ahead of time. If the underlying code switches to use OPTIONS instead of GET, users would start seeing this button initially be clickable but then suddenly disappear.

This is probably a nitpick though and not a blocker. I'm not even sure what the best UX would be.

Reply:
Ah, got it! I wonder if defaulting to disabled would be a better experience...

I think that, further down the track, `core-data` should modify the `canUser` state whenever it receives an `Allow` header from _any_ response, and not just when the `canUser()` resolver is called.

Reply:
Yeah, that's a nice idea. Not sure how it'd work, but a nice idea 😄